src/classes/OPP_OpportunityNamingBTN_CTRL.cls

```diff
@@ -48,14 +49,15 @@ public with sharing class OPP_OpportunityNamingBTN_CTRL {
         list<Opportunity> thisOpp = new list<Opportunity>{(Opportunity)ctrl.getRecord()};
 
         if (!thisOpp.isEmpty()) {
-            
             OPP_OpportunityNaming.refreshOppNames(thisOpp);
-            update thisOpp;
-            
-            PageReference pageRef = new PageReference('/' + thisOpp[0].id);
-            pageRef.setRedirect(true);
-            return pageRef;
-        
-        } else return null;
+            try {
+                update thisOpp;
+                redirect = true;
```

Comment:
this page used to redirect back to the opp page, but now I think it will end up on the empty button page.